velox/common/base/PeriodicStatsReportDaemon.cpp

```diff
@@ -0,0 +1,348 @@
+      "lastSsdCacheCheckpointsRead";
+  static constexpr const char* kLastMemoryCacheAgedOuts =
+      "lastMemoryCacheAgedOuts";
+
+  if (asyncDataCache_ == nullptr) {
+    return;
+  }
+  const auto memoryCacheStats = asyncDataCache_->refreshStats();
+
+  // Snapshots.
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumEntries, memoryCacheStats.numEntries);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumEmptyEntries, memoryCacheStats.numEmptyEntries);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumSharedEntries, memoryCacheStats.numShared);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumExclusiveEntries, memoryCacheStats.numExclusive);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumPrefetchedEntries, memoryCacheStats.numPrefetch);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheTotalTinyBytes, memoryCacheStats.tinySize);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheTotalLargeBytes, memoryCacheStats.largeSize);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheTotalTinyPaddingBytes, memoryCacheStats.tinyPadding);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheTotalLargePaddingBytes, memoryCacheStats.largePadding);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheTotalPrefetchBytes, memoryCacheStats.prefetchBytes);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheSumEvictScore, memoryCacheStats.sumEvictScore);
+
+  // Interval cumulatives.
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumHit,
+      memoryCacheStats.numHit -
+          prevSimpleStatsHolder_[kLastMemoryCacheHits].int64Value);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheHitBytes,
+      memoryCacheStats.hitBytes -
+          prevSimpleStatsHolder_[kLastMemoryCacheHitsBytes].int64Value);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumNew,
+      memoryCacheStats.numNew -
+          prevSimpleStatsHolder_[kLastMemoryCacheInserts].int64Value);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumEvict,
+      memoryCacheStats.numEvict -
+          prevSimpleStatsHolder_[kLastMemoryCacheEvictions].int64Value);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumEvictChecks,
+      memoryCacheStats.numEvictChecks -
+          prevSimpleStatsHolder_[kLastMemoryCacheEvictionChecks].int64Value);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumWaitExclusive,
+      memoryCacheStats.numWaitExclusive -
+          prevSimpleStatsHolder_[kLastMemoryCacheStalls].int64Value);
+  RECORD_METRIC_VALUE(
+      kMetricMemoryCacheNumAllocClocks,
+      memoryCacheStats.allocClocks -
+          prevSimpleStatsHolder_[kLastMemoryCacheAllocClocks].int64Value);
+
+  prevSimpleStatsHolder_[kLastMemoryCacheHits].int64Value =
+      memoryCacheStats.numHit;
+  prevSimpleStatsHolder_[kLastMemoryCacheHitsBytes].int64Value =
+      memoryCacheStats.hitBytes;
+  prevSimpleStatsHolder_[kLastMemoryCacheInserts].int64Value =
+      memoryCacheStats.numNew;
+  prevSimpleStatsHolder_[kLastMemoryCacheEvictions].int64Value =
```

Comment:
How about we just record the previous raw stats instead of using prevSimpleStatsHolder_? Thanks!